client/src/research/LoginPage.js

```diff
@@ -1,11 +1,60 @@
 import React, { Component } from 'react';
+import './LoginPage.css';
 
 
 // The page for users to login for accessing research data.
 class LoginPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: ""
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    // console.log(this);
+    const { value } = e.target;
+    // console.log(e.target);
+    this.setState({ email : value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    fetch('/api/research/login', {
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      method: 'POST',
+      body: JSON.stringify({
+        email: this.state.email
+      })
+    });
+  }
+
+  //Should there be something when email is not valid?
+  //then we'd have to wait for promise to return.....
+  //or make it send an email saying email is not valid?
```

Comment:
Yep, you can factor the `fetch` into its own method like `doSubmitForm`, and then call that method here and add in `.then(this.onSubmitCompleted).catch(this.onSubmitError)`.  For now, you can have those just `console.log` what happened, but what we'll add next is to have them call `this.setState` and add a new bit of state `errorMessage` or something like that, and then update `render` to show the error message if there is one (eg, You're not authorized to sign in).

Reply:
I'm not sure I'm understanding this properly. I'll find you to chat about this sometime!

Reply:
Sounds good!

So here you do want to do something in response to how the server responds.  The `fetch` call will return a promise, and you can `.then` or `.catch` it to see how the server responded.

In the `then` and `catch` blocks, you'll want to show the user what happened.  The way to do that in React is two steps: 1) call `setState` to store an error message, and then 2) update what happens in `render` to show the error message if there is one.

We can chat more in person whenever too!

Reply:
Yeah this looks like a great improvement 👍